rkyv/src/vec/mod.rs

```diff
@@ -178,6 +178,160 @@ impl<T> ArchivedVec<T> {
             Ok(VecResolver { pos })
         }
     }
+
+    /// Serializes an archived `Vec` from a given iterator. Compared to `serialize_from_iter()`,
+    /// this function:
+    /// - supports iterators whose length is not known in advance, and
+    /// - does not collect the data in memory before serializing.
+    ///
+    /// This method will panic if any item writes during `serialize` (i.e no additional data
+    /// written per item).
+    ///
+    /// Usage example:
+    /// ```
+    /// use rkyv::{vec::ArchivedVec, archived_root,
+    ///     ser::{serializers::{WriteSerializer, AllocSerializer}, Serializer},
+    ///     AlignedBytes, Archive, Deserialize, Serialize,
+    /// };
+    ///
+    /// const SCRATCH_SIZE: usize = 256;
+    /// type DefaultSerializer = AllocSerializer<SCRATCH_SIZE>;
+    ///
+    /// // Build some example data structures that are a little bit complex
```

Comment:
Optional: The `Clone` and `Copy` derives aren't required for these types.

Reply:
Copy is not required. `Clone` is required else the doctest fails at compile. 

Fixed in newest commit